frontend/src/components/RecipePage.js

```diff
@@ -0,0 +1,92 @@
+import React, { useEffect, useState } from 'react';
+import { useParams } from 'react-router-dom';
+import Navbar from './navbar';
+import './pages.css';
+
+function RecipePage() {
+
+    let { id } = useParams();
+    const [newIngredient, setNewIngredient] = useState('');//add this
+    const [newStep, setNewStep] = useState('');//add this
+    const [recipe, setRecipe] = useState();
+
+    async function addIngredient() {
+      let ingredient = document.getElementById("newIngredient").value;
+      const rawResponse = await fetch('http://localhost:3001/api/recipe/' + id + '/ingredient', {
+        method: 'PUT',
+        headers: {
+          'Accept': 'application/json',
+          'Content-Type': 'application/json'
+        },
+        body: JSON.stringify({newIngredient : ingredient})
+      });
+      setNewIngredient('');
+    }
+
+    async function addStep() {
+      let step = document.getElementById("newStep").value;
+      const rawResponse = await fetch('http://localhost:3001/api/recipe/' + id + '/instruction', {
+        method: 'PUT',
+        headers: {
+          'Accept': 'application/json',
+          'Content-Type': 'application/json'
+        },
+        body: JSON.stringify({newInstruction : step})
+      });
+      setNewStep('');
+    }
+
+  useEffect(() => {
+    const loadRecipe = () => {
+       fetch('http://localhost:3001/api/recipe/' + id)
+      .then(res => res.json())
+      .then(jsondata => setRecipe(jsondata));
+    }
+    loadRecipe();
+  }, [id, newIngredient, newStep])
```

Comment:
Interesting use of useEffect! This works fine, but I do believe its a bit ineffecient since it makes a fetch request whenever you type into the newIngredient field or newStep field. To optimize, it should only fetch after you click the add ingredient or add step buttons.